src/EnergyPlus/AirflowNetwork/src/Elements.cpp

```diff
@@ -3690,6 +3681,1015 @@ namespace AirflowNetwork {
+        // startheightfactor, heightfactor and ownheightfactor. Thus for
+        // slanted windows the range of the profiles is the vertical
+        // projection of the actual opening.
+
+        // REFERENCES:
+        // Helmut E. Feustel and Alison Rayner-Hooson, "COMIS Fundamentals," LBL-28560,
+        // Lawrence Berkeley National Laboratory, Berkeley, CA, May 1990
+
+        // USE STATEMENTS:
+        // na
+
+        // Argument array dimensioning
+        EP_SIZE_CHECK(DpF, 2);
+        EP_SIZE_CHECK(DpT, 2);
+        EP_SIZE_CHECK(BetaF, 2);
+        EP_SIZE_CHECK(BetaT, 2);
+        EP_SIZE_CHECK(RhoStF, 2);
+        EP_SIZE_CHECK(RhoStT, 2);
+
+        // Locals
+        // SUBROUTINE ARGUMENT DEFINITIONS:
+        // in the FROM zone (starting at linkheight) [Pa]
+        // (starting at linkheight) [Kg/m3]
+        // (starting at linkheight) [Kg/m3]
+
+        // SUBROUTINE PARAMETER DEFINITIONS:
+        // na
+
+        // INTERFACE BLOCK SPECIFICATIONS
+        // na
+
+        // DERIVED TYPE DEFINITIONS
+        // na
+
+        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+        Array1D<Real64> zF(2); // Startheights of layers in FROM-, TO-zone
+        Array1D<Real64> zT(2);
+        Array1D<Real64> zStF(2); // Startheights of layers within the LO, starting with the actual startheight of the LO.
+        Array1D<Real64> zStT(2);
+        // The values in the arrays DpF, DpT, BetaF, BetaT, RhoStF, RhoStT are calculated at these heights.
+        Real64 hghtsFR;
+        Real64 hghtsTR;
+        Array1D<Real64> hghtsF(NrInt + 2); // Heights of evaluation points for pressure and density profiles
+        Array1D<Real64> hghtsT(NrInt + 2);
+        Real64 Interval; // Distance between two evaluation points
+        Real64 delzF;    // Interval between actual evaluation point and startheight of actual layer in FROM-, TO-zone
+        Real64 delzT;
+        int AnzLayF; // Number of layers in FROM-, TO-zone
+        int AnzLayT;
+        int lF; // Actual index for DpF/T, BetaF/T, RhoStF/T, zStF/T
+        int lT;
+        int n;
+        int i;
+        int k;
+
+        // Initialization
+        delzF = 0.0;
+        delzT = 0.0;
+        Interval = ActLh * OwnHeightFactor / NrInt;
+
+        for (n = 1; n <= NrInt; ++n) {
+            hghtsF(n + 1) = state.afn->AirflowNetworkLinkageData(il).NodeHeights[0] + Interval * (n - 0.5);
+            hghtsT(n + 1) = state.afn->AirflowNetworkLinkageData(il).NodeHeights[1] + Interval * (n - 0.5);
+        }
+        hghtsF(1) = state.afn->AirflowNetworkLinkageData(il).NodeHeights[0];
+        hghtsT(1) = state.afn->AirflowNetworkLinkageData(il).NodeHeights[1];
+        hghtsF(NrInt + 2) = state.afn->AirflowNetworkLinkageData(il).NodeHeights[0] + ActLh * OwnHeightFactor;
+        hghtsT(NrInt + 2) = state.afn->AirflowNetworkLinkageData(il).NodeHeights[1] + ActLh * OwnHeightFactor;
+
+        lF = 1;
+        lT = 1;
+        if (From == 0) {
```

Comment:
`?:`?

Reply:
Yes, probably. This is F77++ in it's natural habitat, will need to go over all of this.